### ISSUE TITLE
The UI freezes when I make any changes to pubspec.yaml

### ISSUE THREAD

**Issue by otopba** [pszklarska/FlutterPubVersionChecker#53]
The UI freezes when I make any changes to pubspec.yaml. Even if I just change the version of the application build, my Android Studio freezes for 2-5 seconds. Most of the time it happens when I have a bad internet connection. It feels like dependency checking is happening right on the ui thread. 

I had to disable the plugin

**Comment by sanekyy** [comment]
Same issue

**Comment by alectogeek** [comment]
I have the same issue. Unfortunately, I had to turn the plugin off.

Use case: 
When you have many imports in pubspec file (> 50) the UI freezes for a couple of seconds on any adding/moving library names.

I use MBP 16 M1 Pro.
My Android Studio:
```
Android Studio Arctic Fox | 2020.3.1 Patch 4
Build #AI-203.7717.56.2031.7935034, built on November 21, 2021
Runtime version: 11.0.10+0-b96-7249189 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.0.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 10
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: com.bloc.intellij_generator_plugin, Dart, io.flutter
```

**Comment by Saifallak** [comment]
if pubspec.yaml having many packages, it hangs the while ide.

**Comment by jayjah** [comment]
Well, I forked this project and tried to fix it... I think I solved it by making a minor change to the file `PubPackageInspections.kt`, see here for [reference](https://github.com/jayjah/FlutterPubVersionChecker/commit/949fcf22928468559e971d14a3099fc86c71d82e)

Afterwards I build it locally and installed it with IntelliJ. So far it looks good to me, IDEA is not freezing anymore while checking for dependencies.

I wonder what the author if this package thinks about the minor change...

**Comment by Mocaris** [comment]
> 好吧，我分叉了这个项目，并试图修复它......我想我通过对filePubPackageInspections`PubPackageInspections.kt`进行小幅更改解决了这个问题，请参阅此处[参考](https://github.com/jayjah/FlutterPubVersionChecker/commit/7fe6bfe37412bf4bef2aa62a48d86a9619d9107d)和[参考](https://github.com/jayjah/FlutterPubVersionChecker/commit/949fcf22928468559e971d14a3099fc86c71d82e)>> 之后，我在本地构建它，并使用IntelliJ安装了它。到目前为止，在我看来还不错，IDEA在检查依赖项时不再冻结。>> 我想知道作者如果这个包裹考虑了这个小变化......

you can pull a request

**Comment by Mocaris** [comment]
> Well, I forked this project and tried to fix it... I think I solved it by making a minor change to the file `PubPackageInspections.kt`, see here for [reference](https://github.com/jayjah/FlutterPubVersionChecker/commit/7fe6bfe37412bf4bef2aa62a48d86a9619d9107d) and [reference](https://github.com/jayjah/FlutterPubVersionChecker/commit/949fcf22928468559e971d14a3099fc86c71d82e)
> 
> Afterwards I build it locally and installed it with IntelliJ. So far it looks good to me, IDEA is not freezing anymore while checking for dependencies.
> 
> I wonder what the author if this package thinks about the minor change...


you can pull a request

**Comment by pszklarska** [comment]
Hey @jayjah thanks for your effort to fix this issue! I'll test it over the weekend and push the uodate.

**Comment by alectogeek** [comment]
Hey, folks!
Are there any updates?

**Comment by pszklarska** [comment]
Hey everyone!
Thanks for your patience, I'm still working on that issue. It's a little bit tricky because it probably requires me to re-write a significant part of the plugin, but I hope I'll be able to fix it soon.

**Comment by Saifallak** [comment]
@pszklarska thanks for your hard work.

so this issue is fixed by https://github.com/pszklarska/FlutterPubVersionChecker/pull/56 ?

**Comment by pszklarska** [comment]
Hey @Saifallak! Yes, that issue was fixed in #56. The updated version (1.3.4) is now available to download. Thanks everyone for your help!

I'm closing this issue for now as it should be fixed, but feel free to reopen it if you still encounter this problem.
